### PR TITLE
Overhaul GDB Tutorial

### PR DESCRIPTION
**Pull request by awdeorio**
Overhaul the LLDB tutorial to make the example in sync with recent updates to our IDE tutorials.  Also, recommend TUI mode.

### CODE REVIEW

**Comment by awdeorio** [comment]
Bump @ohjuny :)

File: docs/setup_gdb.md [hunk 1]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
-GDB debugs a compiled executable.  Compile the executable you plan to debug.
+First, compile and run your executable at the command line.
 ```console
-$ pwd
-/Users/awdeorio/src/eecs280/p1-stats
-$ make clean
-rm -rvf *.exe *~ *.out *.dSYM *.stackdump
-$ make stats_tests.exe
-g++ -Wall -Werror -pedantic -g --std=c++11 stats_tests.cpp stats.cpp p1_library.cpp -o stats_tests.exe
+$ make main.exe
```
Comment (ohjuny) [inline docs/setup_gdb.md:60]:
```suggestion
$ make main.exe
g++ -Wall -Werror -pedantic -g --std=c++11 main.cpp -o main.exe
```
I know this command is already shown in the Quick Start snippet above, but I think it's just nice to have the command there for people following along.

File: docs/setup_gdb.md [hunk 2]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
-# Debug
-Start GDB with your executable.
-```console
-$ gdb stats_tests.exe
-GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
-...
-Reading symbols from stats_tests.exe...
-(gdb)
-```
+`n` **Next AKA Step Over**
```
Comment (ohjuny) [inline docs/setup_gdb.md:126]:
```suggestion
`n` **Next / Step Over:**
```
Unsure about AKA here and everywhere else. Thoughts on "Next / Step Over"? Same for all the other AKAs.
Also added colon for visual separateness.

File: docs/setup_gdb.md [hunk 3]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+Run one line of code, stepping _over_ any function calls by running the whole function in one step.
 
-Set a breakpoint on the main function (`b main`) and run the program (`r`).
-```
-(gdb) b main
-Breakpoint 1 at 0x3989: file stats_tests.cpp, line 29.
-(gdb) r
-Starting program: /home/awdeorio/src/eecs280/p1-stats/stats_tests.exe
+`s` **Step AKA Step Into**
```
Comment (ohjuny) [inline docs/setup_gdb.md:129]: Refer to other AKA comment

File: docs/setup_gdb.md [hunk 4]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+Run one line of code, stepping _into_ any function calls to execute them line-by-line.
 
-Breakpoint 1, main () at stats_tests.cpp:29
-29  {
-```
+`up` **Up AKA Step Out**
```
Comment (ohjuny) [inline docs/setup_gdb.md:132]: Refer to other AKA comment

File: docs/setup_gdb.md [hunk 5]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+Run the program until it returns from the current function (or until the next breakpoint).
 
-**Pro-tip**: `start` is a shortcut for `b main` followed by `r`.
+`c` **Continue**
```
Comment (ohjuny) [inline docs/setup_gdb.md:135]:
```suggestion
`c` **Continue:**
```

File: docs/setup_gdb.md [hunk 6]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+Run the program until the next breakpoint.
 
-**Pro-tip**: Other ways to set breakpoints:
+`q` **Quit**
```
Comment (ohjuny) [inline docs/setup_gdb.md:138]:
```suggestion
`q` **Quit:**
```

File: docs/setup_gdb.md [hunk 7]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
 (gdb) n
 (gdb) n
 (gdb) n
-(gdb) refresh
-(gdb)
 ```
+{: data-highlight="11,19-22" }
+
+<div class="primer-spec-callout info" markdown="1">
+**Pro-tip:** Hit Return to repeat your previous command.
```
Comment (ohjuny) [inline docs/setup_gdb.md:308]:
```suggestion
**Pro-tip:** Hit <kbd>Enter</kbd> to repeat your previous command.
```
Guessing Enter is more familiar than Return for Windows users.
The \<kbd> styling is just a thought.

File: docs/setup_gdb.md [hunk 8]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
-Print the value of a variable with `p`.  If you have trouble viewing the contents of the `vector` in the previous step, see the [Pretty Printing STL Containers with `gdb`](#pretty-printing-stl-containers-with-gdb) section.
 ```
 (gdb) p data
-$3 = std::vector of length 3, capacity 4 = {1, 2, 3}
+$1 = std::vector of length 3, capacity 4 = {10,	20, 30}
+```
+{: data-highlight="1" }
+
+### Step into
+Enter `s` AKA "Step" AKA "Step Into".  The cursor enters the `sum()` function.
```
Comment (ohjuny) [inline docs/setup_gdb.md:321]:
```suggestion
Enter `s` (Step / Step Into).  The cursor enters the `sum()` function.
```
Refer to previous AKA comment.

File: docs/setup_gdb.md [hunk 9]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+   │13      int main() {                                                       │
+   └───────────────────────────────────────────────────────────────────────────┘
+native process 95465 In: sum                                 L6    PC: 0x400c02 
+
+(gdb) s
 ```
+{: data-highlight="7,18" }
+
+### Step out
+Enter `up` AKA "Step Out".  The `sum()` function completes, and the program pauses again.
```
Comment (ohjuny) [inline docs/setup_gdb.md:346]:
```suggestion
Enter `up` (Step Out).  The `sum()` function completes, and the program pauses again.
```

File: docs/setup_gdb.md [hunk 10]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+native process 95465 In: main                                L18   PC: 0x400cf0 
 
-  Inferior 1 [process 1090] will be killed.
+(gdb) up
+```
+{: data-highlight="7,18" }
 
-Quit anyway? (y or n) y
+### Continue
+Enter `c` AKA "Continue" to run the program to the next breakpoint, or the end, whichever comes first.
```
Comment (ohjuny) [inline docs/setup_gdb.md:371]:
```suggestion
Enter `c` (Continue) to run the program to the next breakpoint, or the end, whichever comes first.
```

File: docs/setup_gdb.md [hunk 11]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+(gdb) q
 ```
+{: data-highlight="1" }
 
 
-# Pro-tips
+## Pro-tips
 Use the up arrow key to cycle through previously used commands.
 
 Use TAB completion to automatically complete the name of a command or a variable.
```
Comment (ohjuny) [inline docs/setup_gdb.md:393]:
```suggestion
Use <kbd>TAB</kbd> completion to automatically complete the name of a command or a variable.
```

**Review by ohjuny — COMMENTED** [review]
Left a few nits. Don't have a Windows machine though so unable to try running the commands myself.

File: docs/setup_gdb.md [hunk 12]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
-
-Rerun the program.
-```
-(gdb) r
-The program being debugged has been started already.
-Start it from the beginning? (y or n) y
-Starting program: /home/awdeorio/src/eecs280/p1-stats/stats_tests.exe
+double sum (const vector<double> &data) {
+  double total = 0;
+  for (size_t i=0; i<data.size(); ++i) {
```
Comment (ohjuny) [inline docs/setup_gdb.md:151]:
```suggestion
  for (size_t i = 0; i < data.size(); ++i) {
```
nitpicking...

File: docs/setup_gdb.md [hunk 13]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+
+```
+   ┌──main.cpp─────────────────────────────────────────────────────────────────┐
+   │1       #include <iostream>                                                │
+   │2       #include <vector>                                                  │
+   │3       using namespace std;                                               │
+   │4                                                                          │
+   │5       double sum (const vector<double> &data) {                          │
+  >│6         double total = 0;                                                │
+   │7         for (size_t i=0; i<data.size(); ++i) {                           │
```
Comment (ohjuny) [inline docs/setup_gdb.md:331]:
```suggestion
   │7         for (size_t i = 0; i < data.size(); ++i) {                       │
```
Consistency with nitpick above...

File: docs/setup_gdb.md [hunk 14]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+{: data-highlight="1" }
 
 
-# Pro-tips
+## Pro-tips
 Use the up arrow key to cycle through previously used commands.
 
 Use TAB completion to automatically complete the name of a command or a variable.
 
 Use Emacs keyboard shortcuts to enter and edit your current command.
```
Comment (ohjuny) [inline docs/setup_gdb.md:395]:
```suggestion
Use [Emacs](setup_emacs.html) keyboard shortcuts to enter and edit your current command.
```
Consistency with LLDb tutorial

**Review by ohjuny — COMMENTED** [review]
Apologies for all these disjoint reviews; kept finding a few nits while going through the LLDB tutorial after this one

File: docs/setup_gdb.md [hunk 15]
```diff
@@ -8,520 +8,394 @@ GDB Tutorial
...
+</div>
+
+### Inspect
+Print the value of a variable with `p`.  If you have trouble viewing the contents of a `vector`, see the [Pretty Printing STL Containers with `gdb`](#pretty-printing-stl-containers-with-gdb) section.
 
-Print the value of a variable with `p`.  If you have trouble viewing the contents of the `vector` in the previous step, see the [Pretty Printing STL Containers with `gdb`](#pretty-printing-stl-containers-with-gdb) section.
 ```
 (gdb) p data
-$3 = std::vector of length 3, capacity 4 = {1, 2, 3}
+$1 = std::vector of length 3, capacity 4 = {10,	20, 30}
```
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_gdb.md:317]: Good idea.  I added a link.

**Comment by awdeorio** [comment]
I think I address all your comments.  Thanks for the detailed feedback @ohjuny !

**Review by ohjuny — APPROVED** [review]
LGTM 🥮

**Pull request merged**